### PR TITLE
Mmagana/bulk uploader v1/ui punchlist

### PR DESCRIPTION
**Pull request by stereocilia**
You must run `gulp` on the project after this PR to reflect the javascript changes in the app.

  - update how 'host' is set for api requests
  - remove File Archive field
  - enable downloading format and transform yaml definitions

### CODE REVIEW

File: resources/angular/view/Definitions/DefinitionsController.js [hunk 1]
```diff
@@ -18,12 +18,25 @@
         vm.stdFormatString = '';
         vm.availableFormatDefinitions = {};
 
+        $scope.$watch('vm.chosen.right.path', function(newValue, oldValue){
+            if(newValue){
+                vm.chosen.right.fileName = newValue.split('/')[1];
```
Comment (mfitzgibbon-pointblue) [inline resources/angular/view/Definitions/DefinitionsController.js]: Assumes newValue has a "/".  Any chance it wont?

File: resources/angular/view/Definitions/DefinitionsController.js [hunk 2]
```diff
@@ -18,12 +18,25 @@
...
+            }
+        });
+
+        $scope.$watch('vm.chosen.transform.path', function(newValue, oldValue){
+            if(newValue){
+                vm.chosen.transform.fileName = newValue.split('/')[1];
```
Comment (mfitzgibbon-pointblue) [inline resources/angular/view/Definitions/DefinitionsController.js]: same question here

File: resources/angular/view/Definitions/DefinitionsController.js [hunk 3]
```diff
@@ -18,12 +18,25 @@
...
+        });
+
         loadProtocolDefinitions();
 
         function loadStandardFormat(protocol){
             protocol = protocol || vm.chosenProtocol;
-            var standard = findByName("Standard", protocol, vm.availableFormatDefinitions.formats);
-            changeFormat('left', standard);
+            vm.chosen.left = findByName("Standard", protocol, vm.availableFormatDefinitions.formats);
+            vm.chosen.left.fileName = vm.chosen.left.path.split('/')[1];
```
Comment (mfitzgibbon-pointblue) [inline resources/angular/view/Definitions/DefinitionsController.js]: and here...

File: resources/angular/view/Definitions/DefinitionsController.js (hunk 1 above)
Comment (robsptblue) [inline resources/angular/view/Definitions/DefinitionsController.js]: I think the path is always to a subdirectory, correct Tyler? Is there any chance of two levels of directory?  That case would also break.

Comment (thfield) [inline resources/angular/view/Definitions/DefinitionsController.js]: this always looks like "formats/Foo.yaml" or "transforms/Foo.yaml"

File: resources/views/definitions.blade.php [hunk 4]
```diff
@@ -62,14 +62,13 @@
 
             <div class="col-md-6" ng-show="vm.chosen.right.formatString">
                 <h3>Source Format Definition:</h3>
-                <a class="btn btn-default" href="@{{ vm.chosen.right.path }}" target="_blank">Download Format Definition</a>
+                <a class="btn btn-default" href="@{{ vm.chosen.right.path }}" download="@{{ vm.chosen.right.fileName }}">Download Format Definition</a>
```
Comment (thfield) [inline resources/views/definitions.blade.php]: i didn't know about the "download" attribute... nice

File: resources/angular/apiRequest/apiRequest.js [hunk 5]
```diff
@@ -3,10 +3,11 @@
...
         .service('ApiRequestService', ApiRequestService);
 
-    ApiRequestService.$inject = ['$http', 'PreferencesRequest', 'ProjectsRequest', '$interpolate', '$cacheFactory'];
+    ApiRequestService.$inject = ['$http', 'PreferencesRequest', 'ProjectsRequest', '$window', '$interpolate', '$cacheFactory'];
 
-    function ApiRequestService($http, PreferencesRequest, ProjectsRequest, $interpolate, $cacheFactory){
-        var apiHost = 'http://localhost:8080/';
+    function ApiRequestService($http, PreferencesRequest, ProjectsRequest, $window, $interpolate, $cacheFactory){
+        var hostTemplate = $interpolate('http://{{host}}/');
+        var apiHost = hostTemplate({ host:$window.location.host });
```
Comment (thfield) [inline resources/angular/apiRequest/apiRequest.js]: i just made a TODO note today that this should be set dynamically..

**Comment by thfield** [comment]
lgtm, i have the feeling there will be some conflicts with what i've done today on this branch but haven't pushed yet, if you want to go ahead and merge i can deal with them in the morning

**Pull request merged**